lib/note-list/index.tsx

```diff
@@ -357,7 +356,7 @@ export class NoteList extends Component<Props> {
     }
 
     // Deselect the currently selected note if it doesn't match the search query
-    if (filter !== prevProps.filter) {
+    if (searchQuery !== prevProps.searchQuery) {
```

Comment:
looks like this PR needs a rebase against the more recent changes to the `selectedNote`

Reply:
I take it back. looks like we need to circle back here and make sure we don't need this any longer. right now it appears to update the `previousIndex` which eventually should be decoupled from things like the selected note IMO

Reply:
You are very right, this code is no longer needed and could have been the cause of some bugs in searching, good catch!